### PR TITLE
refactor: add usage analytics

### PR DESCRIPTION
**Pull request by allacherkes**
Before submitting a pull request, please ensure the following:

## Description

Please explain the changes you made.

Add a link to the design (if applicable).

### Checklist

- [ ] Code compiles correctly
- [ ] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [ ] Linting run locally successfully
- [ ] All tests pass
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -3,21 +3,26 @@ import { MtaBuildCommand } from "./commands/mtaBuildCommand";
 import { MtarDeployCommand } from "./commands/mtarDeployCommand";
 import { AddModuleCommand } from "./commands/addModuleCommand";
 import { messages } from "./i18n/messages";
-import { createExtensionLoggerAndSubscribeToLogSettingsChanges } from "./logger/logger-wrapper";
+import {
+  createExtensionLoggerAndSubscribeToLogSettingsChanges,
+  getLogger
+} from "./logger/logger-wrapper";
+import { SWATracker } from "@sap/swa-for-sapbas-vsx";
+import * as _ from "lodash";
```
Comment (rayakoren) [inline src/extension.ts:11]: import { partial } from "lodash"

File: tests/extension.spec.ts [hunk 2]
```diff
@@ -27,52 +29,52 @@ describe("Extension unit tests", () => {
   let utilsMock: any;
   let windowMock: any;
   let configSettingsMock: any;
+  let testContext: any;
 
   before(() => {
     sandbox = sinon.createSandbox();
+
```
Comment (rayakoren) [inline tests/extension.spec.ts:36]: beforeEach

File: tests/extension.spec.ts [hunk 3]
```diff
@@ -27,52 +29,52 @@ describe("Extension unit tests", () => {
...
+      .returns("off")
+      .atLeast(1);
+    configSettingsMock
+      .expects("getSourceLocationTrackingSetting")
+      .atLeast(1)
+      .returns(false);
   });
 
   after(() => {
+    configSettingsMock.verify();
```
Comment (rayakoren) [inline tests/extension.spec.ts:55]: afterEach

File: src/extension.ts (hunk 1 above)
Comment (allacherkes) [reply src/extension.ts]: done

File: tests/extension.spec.ts (hunk 2 above)
Comment (allacherkes) [reply tests/extension.spec.ts]: done

File: tests/extension.spec.ts (hunk 3 above)
Comment (allacherkes) [reply tests/extension.spec.ts]: done

**Review by rayakoren — APPROVED** [review]

**Pull request merged**